src/qml/models/options_model.h

```diff
@@ -62,7 +63,10 @@ class OptionsQmlModel : public QObject
     void setUpnp(bool new_upnp);
     QString getDefaultDataDirString();
     QUrl getDefaultDataDirectory();
-    Q_INVOKABLE void setCustomDataDirArgs(QString path);
+    Q_INVOKABLE bool setCustomDataDirArgs(QString path);
+    Q_INVOKABLE QString getCustomDataDirString();
+    bool customDataDir() const { return m_custom_datadir; }
+    void setCustomDataDir(bool new_custom_datadir);
```

Comment:
It seems confusing the current naming convention here, also: do we need to store a bool?; perhaps @johnny9 has a better naming since he has been working with the options model already, mine is just an idea.

```suggestion
    bool isCustomDataDirSet() const { return (!getCustomDataDirString().isEmpty() && (getDefaultDataDirString() != getCustomDataDirString())); }
```

Reply:
yes a bool is needed here so that StorageSettings.qml relfects the proper data directory (either default or custom).

Reply:
ok but do we need that instance variable (`m_custom_datadir`) to store it or it can be dynamically calculated as suggested?

Reply:
Certainly, adhering to the established conventions for options would suggest utilizing m_custom_datadir. However, if a more refined approach emerges, we could potentially explore its implementation in a subsequent pull request.

Reply:
The confusion here is around the name of the variable. customDataDir implies it is a string. A better name for a bool would be isCustomDataDirSet or usingCustomDataDir and customDataDir should just be the name for the string.

Reply:
I prefer the dynamic calculation of the state myself. It is clearer to me

Reply:
Thanks for the suggestions, decided to go with a dynamic string variable instead of managing the customDataDir bool... Let me know your thoughts?